python/src/lib.rs

```diff
@@ -83,6 +84,8 @@ enum PartitionFilterValue<'a> {
 #[pyclass]
 struct RawDeltaTable {
     _table: deltalake::DeltaTable,
+    // storing the config additionally on the table helps us make pickling work.
+    _config: FsConfig,
```

Comment:
Why is that? I wouldn't expect we would need to modify this struct to make the storage handler pickle-able.

Reply:
I went this way to ceated the `DeltaFileSystemHandler` from the delta table instance, since we need the options to be avaialable even if we reuse the underlying object store. An alternative would have been to expose the option on the object store itself, but somehow I felt better not exposing these at the lower level.

A point of concern by the way was, that when pickling, we are writing potentially long lived secrets to disc in plain text. Should we add a comment to that somewhere?

Reply:
Yeah not sure where to leave a comment. But I think if people should implicitly understand that a storage handler contains secrets and they should be careful where they send the pickles.